src/autofix/autofix.ts

```diff
@@ -0,0 +1,376 @@
+	importRequests: ImportRequests;
+	endPos?: number;
+}
+
+function createCompilerHost(sourceFiles: SourceFiles): ts.CompilerHost {
+	return {
+		getSourceFile: (fileName) => sourceFiles.get(fileName),
+		writeFile: () => undefined,
+		getDefaultLibFileName: () => "lib.d.ts",
+		useCaseSensitiveFileNames: () => false,
+		getCanonicalFileName: (fileName) => fileName,
+		getCurrentDirectory: () => "",
+		getNewLine: () => "\n",
+		fileExists: (fileName): boolean => sourceFiles.has(fileName),
+		readFile: () => "",
+		directoryExists: () => true,
+		getDirectories: () => [],
+	};
+}
+
+const compilerOptions = {
+	checkJs: false,
+	allowJs: true,
+	skipLibCheck: true,
+
+	target: ts.ScriptTarget.ES2022,
+	module: ts.ModuleKind.ES2022,
+	isolatedModules: true,
+	sourceMap: true,
+	suppressOutputPathCheck: true,
+	noLib: true,
+	noResolve: true,
+	allowNonTsExtensions: true,
+};
+
+function createProgram(inputFileNames: string[], host: ts.CompilerHost): ts.Program {
+	return ts.createProgram(inputFileNames, compilerOptions, host);
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export default async function ({
+	rootDir: _unused1,
+	namespace: _unused2,
+	resources,
+	context,
+}: AutofixOptions): Promise<AutofixResult> {
+	const sourceFiles: SourceFiles = new Map();
+	const resourcePaths = [];
+	for (const [resourcePath, resource] of resources) {
+		const sourceFile = ts.createSourceFile(
+			resourcePath,
+			resource.content,
+			{
+				languageVersion: ts.ScriptTarget.ES2022,
+				jsDocParsingMode: ts.JSDocParsingMode.ParseNone,
+			}
+		);
+		sourceFiles.set(resourcePath, sourceFile);
+		resourcePaths.push(resourcePath);
+	}
+
+	const compilerHost = createCompilerHost(sourceFiles);
+	const program = createProgram(resourcePaths, compilerHost);
+
+	const checker = program.getTypeChecker();
+	const res: AutofixResult = new Map();
+	for (const [resourcePath, sourceFile] of sourceFiles) {
+		const newContent = applyFixes(checker, sourceFile, resourcePath, resources.get(resourcePath)!, context);
+		if (newContent) {
+			res.set(resourcePath, newContent);
+		}
+	}
+
+	return res;
+}
+
+function applyFixes(
+	checker: ts.TypeChecker, sourceFile: ts.SourceFile, resourcePath: ResourcePath,
+	resource: AutofixResource, context: LinterContext
+): string | undefined {
+	const {content} = resource;
+
+	// Group messages by id
+	const messagesById = new Map<MESSAGE, RawLintMessage[]>();
+	for (const msg of resource.messages) {
+		if (!messagesById.has(msg.id)) {
+			messagesById.set(msg.id, []);
+		}
+		messagesById.get(msg.id)!.push(msg);
+	}
+
+	const changeSet: ChangeSet[] = [];
+	let newModuleDeclarations: NewModuleDeclarationInfo[] = [];
+	if (messagesById.has(MESSAGE.DEPRECATED_API_ACCESS)) {
+		newModuleDeclarations = generateSolutionDeprecatedApiAccess(
+			checker, sourceFile, content,
+			messagesById.get(MESSAGE.DEPRECATED_API_ACCESS) as RawLintMessage<MESSAGE.DEPRECATED_API_ACCESS>[],
+			changeSet);
+	}
```

Comment:
In the first iteration, only globals should be replaced, so this logic should not be in here. I'd be fine with removing the code and coming back to the poc branch at https://github.com/SAP/ui5-linter/tree/add-autofix-poc to add the code again once we tackle migration of deprecated API access / AMD migration.

Reply:
Currently, this code does nothing as we don't have any tests for it.
I can remove it for easier code review, but still have already extracted the code

Reply:
It did nothing in our test scenario, but in real projects it would have tried to do things, which is what we don't want to add for now as a first step.

Reply:
I have cleaned it up